src/Renci.SshNet/Security/KeyExchange.cs

```diff
@@ -255,10 +314,6 @@ public HashAlgorithm CreateClientHash()
                                                Hash(GenerateSessionKey(SharedKey, ExchangeHash, 'E', sessionId)),
                                                _clientHashInfo.KeySize / 8);
 
-            DiagnosticAbstraction.Log(string.Format("[{0}] Creating {1} client hmac algorithm.",
```

Comment:
I'd at least like to discuss why you prefer to remove these. The logging you added in this class does not help if you have multiple key exchanges in progress. The previous logging used a session id prefix for this purpose.

I'd prefer to use that prefix whether it's accessible.